proof/tx.go

```diff
@@ -141,6 +142,11 @@ func (p *TxMerkleProof) Decode(r io.Reader) error {
 	if err != nil {
 		return err
 	}
+
+	if numNodes > math.MaxUint8 {
```

Comment:
I think we should have a named constant in our code instead of `math.MaxUint8` here. Then we can have a comment to explain the constant.

Reply:
Good point. Turns out the value can actually be way lower.